core/src/main/java/org/polypheny/db/catalog/entity/logical/LogicalIndex.java

```diff
@@ -0,0 +1,173 @@
+ * limitations under the License.
+ */
+
+package org.polypheny.db.catalog.entity.logical;
+
+
+import io.activej.serializer.annotations.Deserialize;
+import io.activej.serializer.annotations.Serialize;
+import io.activej.serializer.annotations.SerializeNullable;
+import java.io.Serial;
+import java.io.Serializable;
+import java.util.LinkedList;
+import java.util.List;
+import lombok.EqualsAndHashCode;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.Value;
+import lombok.experimental.SuperBuilder;
+import org.polypheny.db.catalog.Catalog;
+import org.polypheny.db.catalog.entity.PolyObject;
+import org.polypheny.db.catalog.logistic.IndexType;
+import org.polypheny.db.type.entity.PolyBoolean;
+import org.polypheny.db.type.entity.PolyLong;
+import org.polypheny.db.type.entity.PolyString;
+import org.polypheny.db.type.entity.PolyValue;
+import org.polypheny.db.type.entity.numerical.PolyInteger;
+
+
+@EqualsAndHashCode(callSuper = false)
+@Value
+@SuperBuilder(toBuilder = true)
+public class LogicalIndex implements Serializable {
+
+    @Serial
+    private static final long serialVersionUID = -318228681682792406L;
+
+    @Serialize
+    public long id;
+    @Serialize
+    public String name;
+    @Serialize
+    @SerializeNullable
+    public String physicalName;
+    @Serialize
+    public boolean unique;
+    @Serialize
+    public IndexType type;
+    @Serialize
+    public long location; // -1 is Polypheny
+    @Serialize
+    public String method;
+    @Serialize
+    public String methodDisplayName;
+    @Serialize
+    public LogicalKey key;
+    @Serialize
+    public long keyId;
+
+
+    public LogicalIndex(
+            @Deserialize("id") final long id,
+            @Deserialize("name") @NonNull final String name,
+            @Deserialize("unique") final boolean unique,
+            @Deserialize("method") final String method,
+            @Deserialize("methodDisplayName") final String methodDisplayName,
+            @Deserialize("type") final IndexType type,
+            @Deserialize("location") final Long location,
+            @Deserialize("keyId") final long keyId,
+            @Deserialize("key") final LogicalKey key,
+            @Deserialize("physicalName") final String physicalName ) {
+        this.id = id;
+        this.name = name;
+        this.unique = unique;
+        this.method = method;
+        this.methodDisplayName = methodDisplayName;
+        this.type = type;
+        this.location = location;
+        this.keyId = keyId;
+        this.key = key;
+        this.physicalName = physicalName;
+    }
+
+
+    // Used for creating ResultSets
+    public List<LogicalIndexColumn> getCatalogIndexColumns() {
+        int i = 1;
+        List<LogicalIndexColumn> list = new LinkedList<>();
+        for ( String columnName : key.getColumnNames() ) {
+            list.add( new LogicalIndexColumn( id, i++, columnName, this ) );
+        }
+        return list;
+    }
```

Comment:
Index is very relational. However, this is something we could easily also support for the other data models.

Reply:
Same as https://github.com/polypheny/Polypheny-DB/pull/464#discussion_r1518556372